worlds/mm2/__init__.py

```diff
@@ -0,0 +1,314 @@
+    item_name_to_id = lookup_item_to_id
+    location_name_to_id = lookup_location_to_id
+    item_name_groups = item_names
+    location_name_groups = location_groups
+    web = MM2WebWorld()
+    rom_name: bytearray
+    world_version: Tuple[int, int, int] = (0, 3, 1)
+
+    def __init__(self, world: MultiWorld, player: int):
+        self.rom_name = bytearray()
+        self.rom_name_available_event = threading.Event()
+        super().__init__(world, player)
+        self.weapon_damage = deepcopy(weapon_damage)
+
+    def create_regions(self) -> None:
+        menu = MM2Region("Menu", self.player, self.multiworld)
+        self.multiworld.regions.append(menu)
+        for region in mm2_regions:
+            stage = MM2Region(region, self.player, self.multiworld)
+            required_items = mm2_regions[region][0]
+            locations = mm2_regions[region][1]
+            prev_stage = mm2_regions[region][2]
+            if prev_stage is None:
+                menu.connect(stage, f"To {region}",
+                             lambda state, items=required_items: state.has_all(items, self.player))
+            else:
+                old_stage = self.multiworld.get_region(prev_stage, self.player)
+                old_stage.connect(stage, f"To {region}",
+                                  lambda state, items=required_items: state.has_all(items, self.player))
+            stage.add_locations(locations, MM2Location)
+            for location in stage.get_locations():
+                if location.address is None and location.name is not dr_wily:
+                    location.place_locked_item(MM2Item(location.name, ItemClassification.progression,
+                                                       None, self.player))
+            if self.options.consumables in (Consumables.option_1up_etank,
+                                            Consumables.option_all):
+                if region in etank_1ups:
+                    stage.add_locations(etank_1ups[region], MM2Location)
+            if self.options.consumables in (Consumables.option_weapon_health,
+                                            Consumables.option_all):
+                if region in energy_pickups:
+                    stage.add_locations(energy_pickups[region], MM2Location)
+            self.multiworld.regions.append(stage)
+
+    def create_item(self, name: str, force_non_progression: bool = False) -> MM2Item:
+        item = item_table[name]
+        classification = ItemClassification.filler
+        if item.progression and not force_non_progression:
+            classification = ItemClassification.progression_skip_balancing \
+                if item.skip_balancing else ItemClassification.progression
+        if item.useful:
+            classification |= ItemClassification.useful
+        return MM2Item(name, classification, item.code, self.player)
+
+    def get_filler_item_name(self) -> str:
+        return self.multiworld.random.choices(list(filler_item_weights.keys()),
```

Comment:
Avoid multiworld.random
```suggestion
        return self.random.choices(list(filler_item_weights.keys()),
```